tests/conftest.py

```diff
@@ -396,3 +397,98 @@ def log_cli_level(pytestconfig):
+            # and the new thread used for running the IO.
+            # Keeping test_running in a list because bool is immutable.
+            # Locking is used here as a synchronization mechanism to guard
+            # access to test_running.
+            request.test_running = [True]
+
+        results = list()
+
+        def finalizer():
+            """
+            Delete the resources created during setup, used for
+            running IO in the test background
+            """
+            with lock:
+                request.test_running = [False]
+            thread.join()
+            for result in results:
+                log.info(f"IOPs after FIO for pod {request.pod_obj.name}:")
+                log.info(
+                    f"Read: {result.get('jobs')[0].get('read').get('iops')}"
+                )
+                log.info(
+                    f"Write: {result.get('jobs')[0].get('write').get('iops')}"
+                )
+            if hasattr(request, 'pod_obj'):
+                request.pod_obj.delete()
+                request.pod_obj.ocp.wait_for_delete(
+                    resource_name=request.pod_obj.name
+                )
+            if hasattr(request, 'pvc_obj'):
+                request.pvc_obj.delete()
+                request.pvc_obj.ocp.wait_for_delete(
+                    resource_name=request.pvc_obj.name
+                )
+            if hasattr(request, 'sc_obj'):
+                request.sc_obj.delete()
+            if hasattr(request, 'secret_obj'):
+                request.secret_obj.delete()
+
+        request.addfinalizer(finalizer)
+
+        request.secret_obj = helpers.create_secret(
+            interface_type=constants.CEPHFILESYSTEM
+        )
+        request.sc_obj = helpers.create_storage_class(
+            interface_type=constants.CEPHFILESYSTEM,
+            interface_name=helpers.get_cephfs_data_pool_name(),
+            secret_name=request.secret_obj.name
+        )
+        request.pvc_obj = helpers.create_pvc(
+            sc_name=request.sc_obj.name, size='2Gi'
+        )
+        request.pod_obj = helpers.create_pod(
+            interface_type=constants.CEPHBLOCKPOOL,
+            pvc_name=request.pvc_obj.name
+        )
+
+        file_path = os.path.join(
+            request.pod_obj.get_mount_path(), 'fio-rand-readwrite'
+        )
+
+        def run_io_in_bg(state):
+            """
+            Run IO by execution FIO and deleting the file created for FIO on
+            the pod, in a while true loop. Will be running as long as
+            state arg is True. state is an argument passed from the outer
+            fixture function, and it represents the test status
+            (running or not).
+
+            Args:
+                state (list): Contains a bool for the test state
+            """
+            while True:
+                with lock:
+                    if state[0]:
+                        request.pod_obj.run_io('fs', '1G')
```

Comment:
The default runtime is 60 seconds, I think we wil end up starting multiple threads here with this calls, and each one has to be accounted for.

Reply:
A better approach is run fio indefinitely and stop at will, which can be done at test level.

Reply:
only one thread is being created and thread.join takes place in the finalizer above.

Reply:
I will try this in small environment, but as I see each call to pod_obj.run_io invokes a background thread and the last attribute is overwritten.

Reply:
oh I see..you mean here - ocs_ci/ocs/workload.py:49
anyway, why is that a problem?